### ISSUE TITLE
Page.insert_textbox incorrectly handles the last word if it starts a new line

### ISSUE THREAD

**Issue by Mvv9** [pymupdf/PyMuPDF#2637]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
It seems that insert_textbox doesn't handle different fonts properly. Even when i add a default font, the text fits not always perfectly within the textbox. However, with a different font, there's either too much whitespace or the text overflows beyond the textbox.

The while loop will stop because the rc value is usually greater than 0, ( it is the "left white space value" right?) often around 2. However, with some fonts (like Allura), the textbox has almost half of its space left as whitespace.

So the rc value should be much higher.

## To Reproduce (mandatory)
Explain the steps to reproduce the behavior, For example, include a minimal code snippet, example files, etc.

 ```
newdoc[r_frame["page"]-1].insert_font(fontname="Text_font", fontfile="Files/Fonts/" + order["font"])
                
                font="Text_font"
                text_color = (0/255, 38/255, 145/255)

             
                size=order["min-font-size"]
              
                rc= -1
                
         
                while rc < 0 :#or (0 < rc < 10):
                   
                    
                    #print(size)
                    size=size-1
                    rc = newdoc[r_frame["page"]-1].insert_textbox(rect, text,
                        fontsize=size,
                        fontname=font,
                        rotate = 0,
                        color=text_color)
                    print(rc)
```

## Expected behavior (optional)
Describe what you expected to happen (if not obvious).
The text should fit neatly within the rectangle, especially at the bottom. Some whitespace is expected due to varying fonts, but the text shouldn't extend outside the rectangle.

## Screenshots (optional)
See screenshot, for outcome off different fonts.

![Example different fonts](https://github.com/pymupdf/PyMuPDF/assets/47593973/feef7819-b13e-4105-a4b5-c01c220b9a39)

**Comment by JorjMcKie** [comment]
Please provide a script plus the used font to reproduce the problem.
Thanks.

**Comment by Mvv9** [comment]
Example script:

If you make the text longer or shorter, the text will fit, so it's a kind of randomness.
If you use the default font (remove fontname=font from lines 62 & 54), this text will fit perfectly. But if you use this font, it [won't.]

`
import os
import fitz

## page size
rect = (0.0, 0.0, 606.614013671875, 430.8659973144531)

#frame size
r_frame = {'page': 2, 'frames': [fitz.Rect(22.506988525390625, 61.839996337890625, 326.0090026855469, 393.9230041503906)]}

fontfilename = "PremiumUltra21v6.ttf"
fontfileURL = "Fonts/"
template_url = "PDFfile/Base_templates/99bd44b2-8486-4922-a047-762099fe4bbe.pdf"
TemporayFile = "PDFfile/example.pdf"
savefile = "PDFfile/example1.pdf"

doc = fitz.open(template_url)
pageSize = doc[0]
rectSize = pageSize.rect

width = rectSize.width  
height = rectSize.height  
rect = fitz.Rect(0, 0, width, height)

newdoc = fitz.open()
newdoc.new_page(width=rect.width, height=rect.height)
newdoc[-1].set_mediabox(rect)
newdoc.new_page(width=rect.width, height=rect.height)
newdoc[-1].set_mediabox(rect)
newdoc.save(TemporayFile)


def rect_plus(rect):
    space=1
    return  fitz.Rect(rect[0] - space, rect[1] - space, rect[2] + space, rect[3] + space)

#text frame size
rect = (22.506988525390625, 61.839996337890625, 326.0090026855469, 393.9230041503906)
newdoc[r_frame["page"]-1].draw_rect(rect_plus(rect),fill=fitz.utils.getColor('yellow'),  color = (1, 1, 1), width = 0.5, overlay=True,morph=None, stroke_opacity=0, fill_opacity=1, oc=0)


            
text = "The morning sun painted the sky with hues of orange and pink. Birds chirped harmoniously, greeting the new day. Nature awakened, filling the air with life and promise."
font_file_path = os.path.join(fontfileURL, fontfilename)


newdoc[r_frame["page"]-1].insert_font(fontname="F0", fontfile=fontfileURL + fontfilename)

font="F0" 
text_color = (0/255, 38/255, 145/255)

rc = newdoc[r_frame["page"]-1].insert_textbox(rect, text,
fontsize=200, 
fontname=font)
size=50

while rc < 0:

    size=size-1
    rc = newdoc[r_frame["page"]-1].insert_textbox(rect, text,
fontsize=size,
fontname=font,
color=text_color)

newdoc.save(f'{savefile}')
`
[Example files.zip](https://github.com/pymupdf/PyMuPDF/files/12550351/Example.files.zip)

**Comment by JorjMcKie** [comment]
Thank you for the reproduction material.
Found the problem and fixed the problem:
Whenever the last word of a line won't fit the current line buffer and thus must be written orphaned to a new line, the line position pointer is not updated. This will lead to a wrong computation of used-up textbox height.
So there is no relationship to the individual font used. Yout example would e.g. also fail with Helvetica Bold.

**Comment by Mvv9** [comment]
Thanks for your response, this was my first question / post on github ever. So i am new here. If i am right, you working already on a solution? And this bug will fix with the next update?

**Comment by JorjMcKie** [comment]
> Thanks for your response, this was my first question / post on github ever. So i am new here. If i am right, you working already on a solution? And this bug will fix with the next update?

Yes the fix is underway and will be part of the next version.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.4.

**Comment by JorjMcKie** [comment]
The fix has not been applied to the "rebased" implementation only.

**Comment by julian-smith-artifex-com** [comment]
Reopening this, as fix for classic implementation has just been pushed.

Marking this (again) as fixed in next release.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.5.
